Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Collections/BaseObjectCollection.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+using Microsoft.MixedReality.Toolkit.Internal.Definitions.Utilities;
+
+namespace Microsoft.MixedReality.Toolkit.SDK.UX.Collections
+{
+    public abstract class BaseObjectCollection : MonoBehaviour
+    {
+        #region public members
+        /// <summary>
+        /// Action called when collection is updated
+        /// </summary>
+        public Action<BaseObjectCollection> OnCollectionUpdated;
+
+        /// <summary>
+        /// List of objects with generated data on the object.
+        /// </summary>
+        [HideInInspector]
+        public List<ObjectCollectionNode> NodeList = new List<ObjectCollectionNode>();
```

Comment:
Also needs to be a property.

Reply:
Does List<> behave as expected when wrapped in a property? I have the change in a commit -> just not sure about the setter working as one would expect.